semantic-tags/src/Tags/Tagging/Precise.hs

```diff
@@ -43,36 +64,133 @@ firstLine :: Source -> Text
 firstLine = Text.takeWhile (/= '\n') . toText . Source.take 180
 
 
--- FIXME: move GFoldable1 into semantic-ast.
-class GFoldable1 c t where
-  -- | Generically map functions over fields of kind @* -> *@, monoidally combining the results.
-  gfoldMap1
-    :: Monoid b
-    => (forall f . c f => f a -> b)
+-- FIXME: move Traversable1 into semantic-ast.
+-- FIXME: derive Traversable1 instances for TH-generated syntax types.
```

Comment:
If `semantic-ast` were a published hackage package along the lines of `semantic-source`, then we could move `Traversable1` & friends into it, the `tree-sitter-*` packages could depend on it, and we could derive `Traversable1` instances for all of the generated syntax types.

On the other hand, the same would work if we defined this in some _other_ package. `semantic-ast` just seems like a reasonable one to choose.